sdk/c/teaclave_client_sdk.h

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+
+#include <stdarg.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+typedef struct AuthenticationClient AuthenticationClient;
+
+typedef struct FrontendClient FrontendClient;
+
+struct AuthenticationClient *teaclave_connect_authentication_service(const char *address,
+                                                                     const char *enclave_info_path,
+                                                                     const char *as_root_ca_cert_path);
+
+int teaclave_close_authentication_service(struct AuthenticationClient *client);
+
+int teaclave_user_register(struct AuthenticationClient *client,
```

Comment:
You don't need the `struct` keyword in all these functions thanks to the `typedef`.

Also, one C design pattern I've come across elsewhere (in ed448-goldilocks) is to typedef a single element array as `foo_t` while naming the struct `foo_s` so that you can provide typedefs for both use cases while discouraging improper pointer use with uninitialized data. Random example: https://sourceforge.net/p/ed448goldilocks/code/ci/master/tree/src/public_include/decaf/shake.h

Reply:
Thanks for pointing out. This header file is auto generated from the Rust SDK by a tool called `cbindgen` (https://github.com/eqrion/cbindgen). I'm adding a comment in the header to explain this.

I don't write C very often. Learned new from the ed488-goldilocks. Thanks.

Reply:
Ah neat, that looks pretty cool. I'm not a big C developer myself, either, but I was playing around with it the past couple weeks, so the knowledge was fresh :)